### PR TITLE
for src include CMakeLists.txt

### PR DESCRIPTION
**Pull request by AdielDror**
I added the src, include, and CMakeLists files.

### CODE REVIEW

File: include/Animal.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	bool horizontal =  false;
+	bool flag =  false;
+
+
+public:
+	// Constructor and destructor
+	Animal(std::string name);
+	virtual ~Animal()=default;
+
+	struct Location
```
Comment (CPPTesterML) [inline include/Animal.h]: why struct is defined in the class ???
make a separate file for it.

File: src/Animal.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ //Operator +=
+Animal::Location& Animal::Location::operator+=(const Location& rhs)
+{
+	this->row += rhs.row;
+	this->column += rhs.column;
+	return *this;
+}
+
+/* A non-member helper function that checks if the index of moving the animal in the map is valid,
+  if not the animal will stop.*/
```
Comment (CPPTesterML) [inline src/Animal.cpp:62]: consider using of consts for ROWS and COLS

File: src/Zoo.cpp [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+		for (int j = 0; j < 40; j++) {
+			map[i][j] = '-';
+		}
+	}
+
+	Animal::Location loc;
+	int rows = 20;
+	int columns = 40;
+
+	for (int i = 0; i < 3; i++)
```
Comment (CPPTesterML) [inline src/Zoo.cpp]: what is that ????
why using for loop and then checking the index value ??, what is benefit of using the  loop ??
what if you have 10 classes ??

File: include/Zoo.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	void run();
+
+private:
+	// A list that will contain the animal instances
+	std::vector<Animal*> animalList;
+	
+	char map[20][40] = {};
+	
+	// The initial list
+	std::vector<std::vector<std::string>> lst{
```
Comment (CPPTesterML) [inline include/Zoo.h]: why you are using this list ???
you can get these info from the animal class

File: src/Zoo.cpp [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+
+		std::cout << "Enter your choice: ";
+		std::cin >> choice;
+
+		switch (choice)
+		{
+		case 0: break;
+		case 1:
+			try
+			{
```
Comment (CPPTesterML) [inline src/Zoo.cpp]: split each case to function

File: src/Zoo.cpp [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
+	Animal* animal;
+
+
+	int rows = 20;
+	int columns = 40;
+
+	int row = rand() % rows;
+	int column = rand() % columns;
+
+	std::string location = std::to_string(row) + "," + std::to_string(column);
```
Comment (CPPTesterML) [inline src/Zoo.cpp]: you can use operator overloading to print the location why do it manualy ??
define operator<< to print this

**Review by CPPTesterML — CHANGES_REQUESTED** [review]

File: include/Animal.h (hunk 1 above)
Comment (AdielDror) [reply include/Animal.h]: I fixed it.

File: src/Animal.cpp (hunk 2 above)
Comment (AdielDror) [reply src/Animal.cpp:62]: I fixed it.

File: src/Zoo.cpp (hunk 3 above)
Comment (AdielDror) [reply src/Zoo.cpp]: I fixed it.

File: include/Zoo.h (hunk 4 above)
Comment (AdielDror) [reply include/Zoo.h]: I fixed it.

File: src/Zoo.cpp (hunk 5 above)
Comment (AdielDror) [reply src/Zoo.cpp]: I fixed it.

File: src/Zoo.cpp (hunk 6 above)
Comment (AdielDror) [reply src/Zoo.cpp]: I fixed it.

**Pull request merged**